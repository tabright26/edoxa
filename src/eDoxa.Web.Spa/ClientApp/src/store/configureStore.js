import { createStore, compose, applyMiddleware } from "redux";
import { loadUser } from "redux-oidc";
import { loadDoxaTags } from "actions/identity/actionCreators";
import { middleware as thunkMiddleware } from "middlewares/thunkMiddleware";
import { middleware as routerMiddleware } from "middlewares/routerMiddleware";
import { middleware as axiosMiddleware } from "middlewares/axiosMiddleware";
import { middleware as signalrMiddleware } from "middlewares/signalrMiddleware";
import { middleware as loggerMiddleware } from "middlewares/loggerMiddleware";
import userManager from "utils/userManager";
import createRootReducer from "./createRootReducer";

// This enables the webpack development tools such as the Hot Module Replacement.
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default initialState => {
  const store = createStore(createRootReducer(), initialState, composeEnhancers(applyMiddleware(thunkMiddleware, axiosMiddleware, signalrMiddleware, routerMiddleware, loggerMiddleware)));
  loadUser(store, userManager);
  store.dispatch(loadDoxaTags());
  return store;
};
