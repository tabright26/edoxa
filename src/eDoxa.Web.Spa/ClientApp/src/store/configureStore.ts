import { createStore, compose, applyMiddleware } from "redux";
import { loadUser } from "redux-oidc";
import { loadDoxaTags } from "store/doxaTags/actions";
import { middleware as thunkMiddleware } from "middlewares/thunkMiddleware";
import { middleware as routerMiddleware } from "middlewares/routerMiddleware";
import { middleware as axiosMiddleware } from "middlewares/axiosMiddleware";
import { middleware as signalrMiddleware } from "middlewares/signalrMiddleware";
import { middleware as loggerMiddleware } from "middlewares/loggerMiddleware";
import userManager from "utils/userManager";
import rootReducer from "./reducers";
import { RootState } from "./types";

// This enables the webpack development tools such as the Hot Module Replacement.
const composeEnhancers = (window["__REDUX_DEVTOOLS_EXTENSION_COMPOSE__"] as typeof compose) || compose;

export default (initialState: RootState) => {
  const store = createStore(rootReducer, initialState, composeEnhancers(applyMiddleware(thunkMiddleware, axiosMiddleware, signalrMiddleware, routerMiddleware, loggerMiddleware)));
  loadUser(store, userManager);
  const action: any = loadDoxaTags();
  store.dispatch(action);
  return store;
};
