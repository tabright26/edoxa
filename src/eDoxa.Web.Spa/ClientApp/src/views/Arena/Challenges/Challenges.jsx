import React from "react";

import ArenaChallengeIndex from "../../../containers/App/Arena/Challenge/Index";

const Challenges = () => <ArenaChallengeIndex />;

export default Challenges;
