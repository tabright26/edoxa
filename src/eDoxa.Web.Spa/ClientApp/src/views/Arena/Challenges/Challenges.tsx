import React from "react";

import ArenaChallengeIndex from "components/Arena/Challenge/List/List";

const Challenges = () => <ArenaChallengeIndex />;

export default Challenges;
