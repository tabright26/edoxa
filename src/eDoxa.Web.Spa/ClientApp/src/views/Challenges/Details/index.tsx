import React, { FunctionComponent } from "react";

import ChallengeDetails from "components/Service/Challenge/Details";

const Details: FunctionComponent = props => <ChallengeDetails {...props} />;

export default Details;
