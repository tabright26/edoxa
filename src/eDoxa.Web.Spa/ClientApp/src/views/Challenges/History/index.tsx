import React, { FunctionComponent } from "react";

const History: FunctionComponent = () => (
  <h1 className="animated fadeIn mt-4">Challenge History</h1>
);

export default History;
