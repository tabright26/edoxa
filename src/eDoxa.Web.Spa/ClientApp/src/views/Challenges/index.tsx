import React, { FunctionComponent, useEffect } from "react";
import ChallengeList from "components/Service/Challenge/List";
import { loadChallenges } from "store/actions/challenge";
import { connect, MapDispatchToProps } from "react-redux";
import { compose } from "recompose";
import { Row, Col, Card, CardImg, CardImgOverlay } from "reactstrap";
import banner from "assets/img/arena/games/leagueoflegends/banner.jpg";
import large from "assets/img/arena/games/leagueoflegends/large.png";

type OwnProps = {};

type DispatchProps = {
  loadChallenges: () => void;
};

type InnerProps = DispatchProps;

type OutterProps = OwnProps;

type Props = InnerProps & OutterProps;

const Challenges: FunctionComponent<Props> = ({ loadChallenges }) => {
  useEffect((): void => {
    loadChallenges();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <>
      <Row>
        <Col>
          <Card className="my-4">
            <CardImg src={banner} height="200" />
            <CardImgOverlay className="d-flex">
              <img
                className="m-auto"
                alt="leagueoflegends"
                src={large}
                width={320}
                height={125}
              />
            </CardImgOverlay>
          </Card>
        </Col>
      </Row>
      <ChallengeList xs="12" sm="12" md="12" lg="6" />
    </>
  );
};

const mapDispatchToProps: MapDispatchToProps<DispatchProps, OwnProps> = (
  dispatch: any
) => {
  return {
    loadChallenges: () => dispatch(loadChallenges())
  };
};

const enhance = compose<InnerProps, OutterProps>(
  connect(null, mapDispatchToProps)
);

export default enhance(Challenges);
