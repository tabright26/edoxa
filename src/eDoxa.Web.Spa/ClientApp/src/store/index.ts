import { createStore, compose, applyMiddleware } from "redux";

import { RootState } from "store/types";

import { reducer as rootReducer } from "store/reducer";

import { middleware as thunkMiddleware } from "utils/thunk/middleware";
import { middleware as routerMiddleware } from "utils/router/middleware";
import { middleware as axiosMiddleware } from "utils/axios/middleware";
import { middleware as signalrMiddleware } from "utils/signalr/middleware";
import { middleware as loggerMiddleware } from "utils/logger/middleware";

import userManager from "utils/oidc/userManager";

import { loadUser } from "redux-oidc";

// This enables the webpack development tools such as the Hot Module Replacement.
const composeEnhancers =
  (window["__REDUX_DEVTOOLS_EXTENSION_COMPOSE__"] as typeof compose) || compose;

export const configureStore = (initialState: RootState) => {
  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(
      applyMiddleware(
        thunkMiddleware,
        axiosMiddleware,
        signalrMiddleware,
        routerMiddleware,
        loggerMiddleware
      )
    )
  );
  loadUser(store, userManager);
  return store;
};
