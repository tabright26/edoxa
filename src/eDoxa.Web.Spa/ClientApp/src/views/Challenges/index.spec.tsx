import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import Challenges from ".";
import { MemoryRouter } from "react-router-dom";
import store from "store";

it("renders without crashing", () => {
  // Act
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <Challenges />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
