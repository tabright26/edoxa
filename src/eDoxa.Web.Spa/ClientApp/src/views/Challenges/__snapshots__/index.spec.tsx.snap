// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="card"
      >
        <img
          className="card-img"
          height="200"
          src="banner.jpg"
        />
        <div
          className="d-flex card-img-overlay"
        >
          <img
            alt="leagueoflegends"
            className="m-auto"
            height={125}
            src="large.png"
            width={320}
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="row"
  />,
]
`;
